cernopendata/modules/records/search/query.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Open Data Portal.
+# Copyright (C) 2021 CERN.
+#
+# CERN Open Data Portal is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Open Data Portal is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Open Data Portal; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Cernopendata Query factory for REST API."""
+
+from elasticsearch_dsl.query import Q
+from flask import current_app, request
+from invenio_records_rest.errors import InvalidQueryRESTError
+from invenio_records_rest.sorter import default_sorter_factory
+from invenio_records_rest.facets import default_facets_factory
+
+
+def cernopendata_search_factory(self, search, query_parser=None):
+    """Customized parse query using invenio query parser.
+
+    :param self: REST view
+    :param search: Elastic search DSL search instance
+
+    :return: Tuple with search instance and URL arguments
+    """
+    def _default_parser(_query_string=None):
+        """Default parser that uses the Q() from elasticsearch_dsl."""
+        if '"' not in _query_string:
+            _query_string = _query_string.replace("/", "//")
```

Comment:
This fixes the internal error for regexp-like queries, but it also leads to some unexpected findings, such as:

- searching for `10.7483/OPENDATA.9S5F.BY3B` will find `/glossary/Barn` as one of the matching hits.  Do you know why was this matched?  

- searching for `"10.7483/OPENDATA.9S5F.BY3B"` will find hundreds of results which are not related, like `record/349`.

An alternative approach could be to quote expressions... We can see IRL

Reply:
Here is a table showing an overview of the above findings:

query | opendata.cern.ch (production) | opendata-dev.cern.ch (master) | PR 3054
----- | ----------------------------- | ----------------------------- | -------
10.7483/OPENDATA.9S5F.BY3B | error | 357 results | 359 results
"10.7483/OPENDATA.9S5F.BY3B" | 1 result | error | 5444 results 
10.7483 | 2123 results | 357 results | 359 results 

Basically the PROD system has the best result, the only thing we want to correct is the "error"  when people don't use quotes.  The PR fixes the error, but finds too much hits for quoted query, and not enough hits for partial query.

Reply:
Yes, I am currently testing this one.

Reply:
Ditto for CMS dataset title search; while the PR fixes the error situations, it also finds way too many "unrelated" results:

query | opendata.cern.ch (production) | opendata-dev.cern.ch (master) | PR 3054
----- | ----------------------------- | ----------------------------- | -------
/BTau/Run2010B-Apr21ReReco-v1/AOD | error | error | 4851 results
"/BTau/Run2010B-Apr21ReReco-v1/AOD" | 2 results | error | 4851 results 
BTau | 5 results | 5 results | 5 results

Reply:
Ping @tiborsimko